packages/myst-cli/src/transforms/code.spec.ts

```diff
@@ -114,3 +114,93 @@ describe('liftCodeMetadataToBlock', () => {
     expect(mdast.children[0].children[1].value).toEqual('print("hello world2")');
   });
 });
+
+function build_mdast(tags: string[], has_output: boolean) {
+  const mdast: any = {
+    type: 'root',
+    children: [
+      {
+        type: 'block',
+        children: [
+          {
+            type: 'code',
+            executable: true,
+          },
+        ],
+        data: {
+          tags: tags,
+        },
+      },
+    ],
+  };
+  if (has_output) {
+    mdast.children[0].children.push({ type: 'output' });
+  }
+  return mdast;
+}
+
+// describe('checkMetaTags', () => {
+//   it('duplicate tag warn', async () => {
+//     const consoleSpy = jest.spyOn(console, 'warn');
+//     for (const action of ['hide', 'remove']) {
+//       for (const target of ['cell', 'input', 'output']) {
+//         const tag = `${action}-${target}`;
+//         const mdast = build_mdast([tag, tag], true);
+//         propagateBlockDataToCode(new Session(), '', mdast);
+//         expect(consoleSpy).toHaveBeenCalledWith(`tag '${tag}' is duplicated`);
```

Comment:
I'm not quite sure how mystjs do warning assertion test. The standard jest spy strategy seems not to work with the session warning. The error report is weird also, the Expected and Received actually have been the same.
```
 FAIL  src/transforms/code.spec.ts (6.465 s)
  propagateBlockDataToCode
    ✕ duplicate tag warn (32 ms)

  ● propagateBlockDataToCode › duplicate tag warn

    expect(jest.fn()).toHaveBeenCalledWith(...expected)

    Expected: "tag 'hide-cell' is duplicated"
    Received: "tag 'hide-cell' is duplicated"

    Number of calls: 1
```